aws/resource_aws_sagemaker_notebook_instance.go

```diff
@@ -0,0 +1,390 @@
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"security_groups": {
+				Type:     schema.TypeSet,
+				MinItems: 1,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"kms_key_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsSagemakerNotebookInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	name := d.Get("name").(string)
+
+	createOpts := &sagemaker.CreateNotebookInstanceInput{
+		SecurityGroupIds:     expandStringSet(d.Get("security_groups").(*schema.Set)),
+		NotebookInstanceName: aws.String(name),
+		RoleArn:              aws.String(d.Get("role_arn").(string)),
+		InstanceType:         aws.String(d.Get("instance_type").(string)),
+	}
+
+	if s, ok := d.GetOk("subnet_id"); ok {
+		createOpts.SubnetId = aws.String(s.(string))
+	}
+
+	if k, ok := d.GetOk("kms_key_id"); ok {
+		createOpts.KmsKeyId = aws.String(k.(string))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		tagsIn := v.(map[string]interface{})
+		createOpts.Tags = tagsFromMapSagemaker(tagsIn)
+	}
+
+	log.Printf("[DEBUG] sagemaker notebook instance create config: %#v", *createOpts)
+	_, err := conn.CreateNotebookInstance(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating Sagemaker Notebook Instance: %s", err)
+	}
+
+	d.SetId(name)
+	log.Printf("[INFO] sagemaker notebook instance ID: %s", d.Id())
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			sagemaker.NotebookInstanceStatusUpdating,
+			sagemaker.NotebookInstanceStatusPending,
+			sagemaker.NotebookInstanceStatusStopped,
+		},
+		Target:  []string{sagemaker.NotebookInstanceStatusInService},
+		Refresh: sagemakerNotebookInstanceStateRefreshFunc(conn, d.Id()),
+		Timeout: 10 * time.Minute,
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("error waiting for sagemaker notebook instance %q to create: %s", d.Id(), err)
+	}
+
+	return resourceAwsSagemakerNotebookInstanceRead(d, meta)
+}
+
+func resourceAwsSagemakerNotebookInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	// TODO change this to describeNotebook Instance
+	describeNotebookInput := &sagemaker.DescribeNotebookInstanceInput{
+		NotebookInstanceName: aws.String(d.Id()),
+	}
+	notebookInstance, err := conn.DescribeNotebookInstance(describeNotebookInput)
+	if err != nil {
+		if isAWSErr(err, "", "RecordNotFound") {
+			d.SetId("")
+			log.Printf("[LOG] Unable to find sageMaker notebook instance %q; removing from state file", d.Id())
+			return nil
+		}
+		return fmt.Errorf("error finding sagemaker notebook instance %q: %s", d.Id(), err)
+
+	}
+
+	d.Set("security_groups", flattenStringList(notebookInstance.SecurityGroups))
```

Comment:
Should add error checking here to match everything else 👍 

```suggestion
	if err := d.Set("security_groups", flattenStringList(notebookInstance.SecurityGroups)); err != nil {
		return fmt.Errorf("error setting security_groups for sagemaker notebook instance %q: %s", d.Id(), err)
	}
```